hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/HermesServer.java

```diff
@@ -151,4 +159,23 @@ private HttpHandler handlers() {
     private boolean isFrontendRequestDumperEnabled() {
         return hermesServerParameters.isRequestDumperEnabled();
     }
+
+    @VisibleForTesting
+    @SuppressWarnings("unchecked")
+    public int getPort() {
+        Field channelsField = FieldUtils.getDeclaredField(Undertow.class, "channels", true);
```

Comment:
Do we need to use reflection to read Undertow port? According to a [thread](https://stackoverflow.com/questions/55678642/unable-to-fetch-random-port-when-server-port-is-0) on Stackoverflow something like this should work:

```
InetSocketAddress socketAddress = (InetSocketAddress) undertow.getListenerInfo().stream().findFirst().orElseThrow(() -> new IllegalStateException("No port available yet.")).getAddress();
return socketAddress.getPort();
```

Reply:
Good point. I applied your suggestion. Thanks.